docs/reduction-in-locks.md

```diff
@@ -0,0 +1,75 @@
+
+If the backup lock is disabled with the [`--lock-ddl=OFF`](./xtrabackup-option-reference.md/#lock-ddl) option, a backup continues while concurrent DDL events are executed. These backups may be invalid and may fail at either the `backup` or the `--prepare` step.
+
+## The `--lock-ddl=REDUCED` option overview
+
+[Percona XtraBackup 8.4.0-2](./release-notes/8.4.0-2.md) adds the [`--lock-ddl=REDUCED`](./xtrabackup-option-reference.md/#lock-ddl) option to reduce the time the server remains under the backup lock. Now, you can execute more `DDL` operations concurrently while the backup is in progress. The backup lock is now taken after copying the `.ibd` files and before copying the `non-InnoDB` files.
+
+Copying the InnoDB files with the `--lock-ddl=REDUCED` option is split into two phases.
+
+### Phase 1: `Operations performed without the lock`
+
+1. Copying all redo logs from checkpoint up to the current LSN and start following new entries.
+2. Starting the redo log thread.
+3. Tracking file operations from the redo log.
+4. Copying of all .ibd without taking any lock.
+5. Acquiring Lock Tables For Backup (LTFB)/Lock Instance For Backup (LIFB). This step ensures no new DDL operations, such as creating or altering tables, will occur.
+6. Quering the log_status to discover the LSN. This step ensures that all subsequent operations are tracked and that the backup process captures a consistent state of the database.
+7. Copying all InnoDB files.
+8. Ensuring the redo log has catch up to LSN from step 5.
+
+### Phase 2: `Operations performed under the lock`
+
+9. Checking the file operations that were tracked and recopying the tablespaces.
+10. Creating additional `meta` files to perform the required actions (deletions or renames) on the already copied files. This approach ensures that the backup remains consistent and accurate without disrupting the streaming process. This step is required for taking streaming backups.
+11. Gathering a sync point from all engines (InnoDB LSN, binlog, GTID, etc.) by querying the `log_status`.
+12. Stoping the redo follow thread once it copies at least up to sync point at step 11.
+13. Releasing LTFB/LIFB.
+14. Processing the new files during `--prepare` phase before crash recovery starts.
+
+### Limitations
+
+1. The `ALTER INSTANCE ROTATE MASTER KEY` command should not be executed during [Phase 1](#phase-1-operations-performed-without-the-lock) if there are encrypted tables, as it will cause the backup to fail.
+
+2. The number of open file handles in your Operation System (OS) should be configured to match the number of files in the backup directory.
+
+3. When using the `--lock-ddl=REDUCED` option, backups can be larger in size. This happens because there are more operations allowed during backups.
+
+4. When the `DDL` operations that skip redo (for example, ADD INDEX) or change encryption types are executed while the backup is in progress, the `xtrabackup` can copy such tablespaces twice.
```

Comment:
These two points can be combined into a single point.

Backups could be larger because new DDL operations are concurrently executed on the server, and these files are copied to the backup. Also, some DDLs, like ADD INDEX, encrypting changes to existing data files would cause a recopy of the data files.

Reply:
@alina-derkach-oaza sorry, I just updated the last line of the above comment a bit

Reply:
No problems:)

Reply:
updated